app/models/concerns/people/membership_verification.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2024, Schweizer Alpen-Club. This file is part of
+#  hitobito and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito.
+
+module People::MembershipVerification
+  extend ActiveSupport::Concern
+
+  def membership_verify_token
+    (new_record? || super.present?) ? super : init_membership_verify_token
+  end
+
+  # token should not be set manually
+  def membership_verify_token=(_value); end
+
+  private
+
+  def init_membership_verify_token
+    token = SecureRandom.base58(24)
+    if Person.where(membership_verify_token: token).exists?
+      raise 'token must be unique'
+    end
```

Comment:
Das kann in diversen Features auftreten und unvorhersehbare Folgen haben. Besser mit retry:
```suggestion
    while Person.where(membership_verify_token: token).exists?
      token = SecureRandom.base58(24)
    end
```

Reply:
Du hast das raise behalten, dadurch ist es noch nicht besser als vorher. Bitte ersetze das raise durch die retry-Logik

Reply:
ah, itsä schnallis, merci !